cirq/sim/mps_simulator_test.py

```diff
@@ -0,0 +1,196 @@
+            )
+        )
+        == "measurements: m=1\n"
+        "output state: [array([[[1., 0.]]])]"
+    )
+
+
+def test_empty_step_result():
+    q0 = cirq.LineQubit(0)
+    state = cirq.MPSState(qubit_map={q0: 0})
+    step_result = cirq.MPSSimulatorStepResult(state, measurements={'0': [1]})
+    assert str(step_result) == "0=1\n[array([[[1., 0.]]])]"
+
+
+def test_state_equal():
+    q0, q1 = cirq.LineQubit.range(2)
+    state0 = cirq.MPSState(qubit_map={q0: 0})
+    state1 = cirq.MPSState(qubit_map={q1: 0})
+    assert state0 == state0
+    assert state0 != state1
+
+
+def test_simulate_moment_steps_sample():
+    q0, q1 = cirq.LineQubit.range(2)
+    circuit = cirq.Circuit(cirq.H(q0), cirq.CNOT(q0, q1))
+
+    simulator = cirq.MPSSimulator()
+
+    for i, step in enumerate(simulator.simulate_moment_steps(circuit)):
+        if i == 0:
+            np.testing.assert_almost_equal(
+                step._simulator_state().to_numpy(),
+                np.asarray([1.0 / math.sqrt(2), 0.0, 1.0 / math.sqrt(2), 0.0]),
+            )
+            assert str(step) == "[array([[[0.70710678+0.j, 0.70710678+0.j]]]), array([[[1., 0.]]])]"
+            samples = step.sample([q0, q1], repetitions=10)
+            for sample in samples:
+                assert np.array_equal(sample, [True, False]) or np.array_equal(
+                    sample, [False, False]
+                )
+            np.testing.assert_almost_equal(
+                step._simulator_state().to_numpy(),
+                np.asarray([1.0 / math.sqrt(2), 0.0, 1.0 / math.sqrt(2), 0.0]),
+            )
+        else:
+            np.testing.assert_almost_equal(
+                step._simulator_state().to_numpy(),
+                np.asarray([1.0 / math.sqrt(2), 0.0, 0.0, 1.0 / math.sqrt(2)]),
+            )
+            assert (
+                str(step)
+                == """[array([[[0.84089642+0.j, 0.        +0.j],
+        [0.        +0.j, 0.84089642+0.j]]]), array([[[0.84089642+0.j, 0.        +0.j]],
+
+       [[0.        +0.j, 0.84089642+0.j]]])]"""
+            )
+            samples = step.sample([q0, q1], repetitions=10)
+            for sample in samples:
+                assert np.array_equal(sample, [True, True]) or np.array_equal(
+                    sample, [False, False]
+                )
+
+
+def test_sample_seed():
+    q = cirq.NamedQubit('q')
+    circuit = cirq.Circuit(cirq.H(q), cirq.measure(q))
+    simulator = cirq.MPSSimulator(seed=1234)
+    result = simulator.run(circuit, repetitions=20)
+    measured = result.measurements['q']
+    result_string = ''.join(map(lambda x: str(int(x[0])), measured))
+    assert result_string == '01011001110111011011'
+
+
+def test_run_no_repetitions():
+    q0 = cirq.LineQubit(0)
+    simulator = cirq.MPSSimulator()
+    circuit = cirq.Circuit(cirq.H(q0), cirq.measure(q0))
+    result = simulator.run(circuit, repetitions=0)
+    assert sum(result.measurements['0']) == 0
```

Comment:
Could we instead test `len(...) == 0`? Using `sum` seems like it could hide strange behavior or cause flakes - for example, if `repetitions=0` actually ran one repetition, but the measurement result was a zero.

Reply:
Done.